src/librustc_lint/builtin.rs

```diff
@@ -349,7 +349,7 @@ impl MissingDoc {
         id: Option<hir::HirId>,
         attrs: &[ast::Attribute],
         sp: Span,
-        desc: &'static str,
+        desc: &str,
```

Comment:
If you pass the `article` and `desc` separately you can avoid needing to allocate a string ahead of time (and you could keep the `'static`).